.github/workflows/reusable-build.yaml

```diff
@@ -0,0 +1,69 @@
+name: "Build Contracts"
+
+on:
+  workflow_call:
+    inputs:
+      commit-hash:
+        description: "The commit hash to build from"
+        type: string
+        required: true
+    outputs:
+      artifact-name:
+        description: "Name of the uploaded artifact containing all builds"
+        value: ${{ jobs.build.outputs.artifact-name }}
+
+jobs:
+  build:
+    runs-on: blacksmith-8vcpu-ubuntu-2204
+    outputs:
+      artifact-name: ${{ steps.set-artifact-name.outputs.name }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Checkout specific commit
+        run: git checkout ${{ inputs.commit-hash }}
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.81.0
```

Comment:
create a separate ticket to read this from the root Cargo.toml for all workflows. Avoids the need to hardcode